File: app/src/main/java/com/hqq/core_app/ui/activity/fragment/MainFragment.java
package com.hqq.core_app.ui.activity.fragment;

import android.os.Bundle;
import android.os.Handler;
import android.support.v4.app.Fragment;

import com.hqq.core.ui.BaseListFragment;
import com.hqq.core_app.R;
import com.hqq.core_app.adapter.MainAdapter;
import com.hqq.core_app.bean.MainBean;
import com.hqq.core_app.ui.activity.IFragmentActivity;
import com.hqq.core_app.ui.activity.bar.ToolBarActivity;

import java.util.ArrayList;
import java.util.List;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.iblibrary.ui.activity.fragment
 * @FileName :   MainFragment
 * @Date : 2018/11/23 0023  上午 9:41
 * @Descrive :
 * @Email :  qiqiang213@gmail.com
 */
public class MainFragment extends BaseListFragment<MainAdapter> {

    @Override
    public int setViewId() {
        return R.layout.fragment_i;

    }

    public static Fragment getIFragment(int position) {
        MainFragment fragment = new MainFragment();
        Bundle bundle = new Bundle();
        bundle.putInt("postition", position);
        fragment.setArguments(bundle);
        return fragment;
    }

    /**
     * 只有界面显示的时候才会加载
     *
     * @return
     */
    @Override
    public boolean isLazyLoad() {
        return true;
    }

    @Override
    public boolean isShowLoadMore() {
        return true;
    }

    @Override
    public MainAdapter initAdapter() {
        return new MainAdapter();

    }

    @Override
    public void initData() {
        initlistData();
    }

    @Override
    protected void onLoadMore() {
        super.onLoadMore();
        getData();
    }

    private void getData() {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                initlistData();
            }
        }, 3 * 1000);

    }

    private void initlistData() {
        List<MainBean> list = new ArrayList<>();
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        list.add(new MainBean("ItoolBar 控制 ", ToolBarActivity.class));
        list.add(new MainBean("fragment 加载", IFragmentActivity.class));
        mBaseListModel.fillingData(list);
    }
}

File: app/src/main/java/com/easy/example/ui/fragment/MainFragment.kt
package com.easy.example.ui.fragment

import android.os.Bundle
import android.os.Handler
import androidx.fragment.app.Fragment
import com.easy.core.ui.list.BaseListFragment
import com.easy.example.R
import com.easy.example.adapter.MainAdapter
import com.easy.example.bean.MainBean
import com.easy.example.ui.bar.ToolBarActivity
import com.easy.example.ui.view.page.IFragmentActivityBuilder
import kotlin.collections.ArrayList

/**
 * @Author : huangqiqiang
 * @Package : com.easy.iblibrary.ui.activity.fragment
 * @FileName :   MainFragment
 * @Date : 2018/11/23 0023  上午 9:41
 * @Descrive :
 * @Email :  qiqiang213@gmail.com
 */
class MainFragment(override val layoutViewId: Int = R.layout.fragment_i,
                   override val isLazyLoad: Boolean = true,//只有界面显示的时候才会加载
                   override val adapter: MainAdapter = MainAdapter()) : BaseListFragment() {

    companion object {
        @JvmStatic
        fun getIFragment(position: Int): Fragment {
            val fragment = MainFragment()
            val bundle = Bundle()
            bundle.putInt("postition", position)
            fragment.arguments = bundle
            return fragment
        }
    }

    override fun initData() {
        Handler().postDelayed({ initListData() }, 3 * 1000.toLong())

    }

    override fun onLoadMore() {
        super.onLoadMore()
        data
    }

    private val data: Unit
        get() {
            Handler().postDelayed({ initListData() }, 3 * 1000.toLong())
        }

    private fun initListData() {
        val list: MutableList<com.easy.example.bean.MainBean<*>> = ArrayList()
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        list.add(com.easy.example.bean.MainBean("ItoolBar 控制 ", ToolBarActivity::class.java))
        list.add(com.easy.example.bean.MainBean("fragment 加载", IFragmentActivityBuilder::class.java))
        listModel.fillingData(list as ArrayList<*>)
    }



}